tests/tokenization/test_tokenization_fast.py

```diff
@@ -240,3 +240,39 @@ def test_async_share_tokenizer(self):
 
     def fetch(self, tokenizer, text):
         return tokenizer.encode(text, truncation="longest_first", padding="longest")
+
+
+@require_tokenizers
+class SplitSpecialTokensTests(unittest.TestCase):
+    def test_split_special_tokens(self):
+        tokenizer = PreTrainedTokenizerFast.from_pretrained('robot-test/dummy-tokenizer-fast', split_special_tokens=True)
```

Comment:
not super necessary, if we want to test this feature for all models we need to add a test that is a bit more general. We could simply compare fast and slow output, adding `<my_new_token>` and making sure that this token is not in the output !

Reply:
good call! Something like this? :

sample_special_token = tokenizer.all_special_tokens[0]  # always exists?
text = f'Here is an example of special token: {sample_special_token}'
tokens_split = tokenizer.tokenize(text)
self.assertNotIn(sample_special_token, tokens_split)
        
what do you mean by compare fast and slow output? combine the tests in _fast and _utils?

Reply:
```python3
tokenizer.add_tokens(["<my_new_token>"])
out = tokenzier.tokenize("Hey how are you<my_new_token>")
assert "<my_new_token>" not in out
```
something like that, which we do for both the fast and slow. Examples of how to test for both here: https://github.com/huggingface/transformers/blob/39114c03838f9cb0c8805b64366d534007021d85/tests/models/t5/test_tokenization_t5.py#L273-L298 😉